### ISSUE TITLE
Global setup for Custom selector engines isn't working

### ISSUE THREAD

**Issue by KingNero9** [microsoft/playwright#20281]
**Context:**
- Playwright Version: Version 1.21.1
- Operating System: Mac
- Node.js version: v12.22.12
- Browser: Chromium
- Extra: React 

Hi! 
I'm trying to create a global setup for custom selector in my playwright tests (react), and i keep getting the following error: **locator.waitFor: Unknown engine "data-automation-id" while parsing selector data-automation-id=componentName.**
Like you can probably understand, i'm looking for a component with custom the html tag of 'data-automation-id="componentName"'.

currently, my files look like this:

**Code Snippet**
// playwright.config.js
```javascript
const config = {
  globalSetup: require.resolve('./global-setup'),
  ...
};
module.exports = config;
```

// global-setup.js
```javascript
const { selectors } = require('@playwright/test');
const createDataAutomationIdSelector = require('..../customSelectors');

module.exports = async () => {
  selectors.register('data-automation-id', createDataAutomationIdSelector);
};
```

// customSelectors.js
```javascript
function createDataAutomationIdSelector() {
  return {
    query(root, selector) {
      return root.querySelector([data-automation-id="${selector}"]);
    },
    queryAll(root, selector) {
      return Array.from(root.querySelectorAll([data-automation-id="${selector}"]));
    },
  };
}

module.exports = createDataAutomationIdSelector;
```

The imports and exports are correct. 
i've tried various variations of those files and nothing seemed to work.. 
But, if i wrote this line in some random file :  selectors.register('data-automation-id', createDataAutomationIdSelector);
and assured it ran only once in the application, the selectors did manage to get registered. and everything passed.

I'm following those two documentations :
https://playwright.dev/docs/test-advanced#configuration-object
https://playwright.dev/docs/extensibility

**Comment by pavelfeldman** [comment]
Your global setup only works in the main process of the test runner, does not work in any of the workers that actually run the tests. You should always register custom engines from your config or from your tests, not from the global setup.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.

**Comment by KingNero9** [comment]
@pavelfeldman thanks for the info! 
Do you have an example for an implementation of custom engines inside the playwright.config? I can't make it work that way.

**Comment by rubenhelsloot** [comment]
@pavelfeldman Could you maybe provide an example of how to register custom engines inside the `playwright.config` file? Thanks in advance!

**Comment by luixo** [comment]
Registering custom engine inside `playwright.config.ts` is simple as that:
```ts
import { selectors } from "@playwright/test";
selectors.register("myTag", () => ({
  query: ...,
  queryAll: ...
}));
```
